cli/main.py

```diff
@@ -29,201 +115,184 @@ def auth() -> str:
 
-    resp = session.get(
-        endpoint,
-    )
-    try:
-        resp.raise_for_status()
-    except:
-        print(resp.json())
-        raise
-
-    print(json.dumps(resp.json()))
+    resp = session.get(endpoint)
+    _output_result(resp)
 
 
 @cli.command()
 @click.pass_context
-@click.option('--project', '-p', required=True, help='ID of the test project to use')
-@click.option('--test-suite', '-t', required=True, help='Test suite to use')
-@click.option('--subscription', '-s', required=True, help='Internal penguinator ID of the "subscription" to use')
-@click.option('--marketplace-image-urn', '-u', type=str, help='URN of the image ("az vm image list -p Canonical --all")')
-@click.option('--vhd-sas-url', '-v', type=str, help='SAS URL of a VHD to test')
-@click.option('--vm-generation', '-g', default=2, type=int, help="Hyper-V generation (1 or 2)")
-@click.option('--vm-size', '-m', help="VM size")
-def create_job(ctx, project: str, test_suite: str, subscription: str, marketplace_image_urn: str, vhd_sas_url: str, vm_generation: int, vm_size: str):
+@_common_options
+@click.option("--name", "-n", required=True, help="job name")
+@click.option(
+    "--marketplace-image-urn",
+    "-u",
+    help='URN of the image ("az vm image list -p Canonical --all")',
+)
+@click.option("--vhd-sas-url", "-v", type=str, help="SAS URL of a VHD to test")
+@click.option("--architecture", "-a", type=click.Choice(['x64', 'arm64']), help="Architecture of the image")
+@click.option(
+    "--vm-generation", "-g", default=2, type=int, help="Hyper-V generation (1 or 2)"
+)
+@click.option(
+    "--region", "-r", default=["westeurope"], type=str, multiple=True, help="provisioning region for test resources"
+)
+@click.option("--template-name", "-t", type=str, help="job template name")
+@_template_options
+def create_job(
+    ctx,
+    name: str,
+    marketplace_image_urn: str,
+    vhd_sas_url: str,
+    architecture: str,
+    template_name: str,
+    vm_generation: int,
+    vm_size: str,
+    priority: str,
+    case_name: str,
+    location: str,
+    region: list[str],
+    **kwargs,
+):
     """
     Create a new test job
     """
-    endpoint = urllib.parse.urljoin(API_BASE_URL, "jobs")
+    endpoint = _get_endpoint(f"jobs/{name}", **kwargs)
 
-    session = ctx.obj['session']
-    payload = {
-        'test_project': project,
-        'test_suite': test_suite,
-        'azure_subscription': subscription,
-        'vm_generation': vm_generation,
-    }
+    session = ctx.obj["session"]
 
+    image = {
+        "vhdGeneration": vm_generation,
+        "architecture": architecture,
+    }
     if marketplace_image_urn:
-        payload['marketplace_image_urn'] = marketplace_image_urn
+        image["type"] = "marketplace"
+        image_parts = marketplace_image_urn.split(":")
+        if len(image_parts) != 4:
+            raise ValueError(
+                "marketplace_image_urn should be in the format of 'publisher:offer:sku:version'"
```

Comment:
It would be helpful for the error to include the value that the function believes to be the URN.